python/triton/runtime/autotuner.py

```diff
@@ -111,7 +111,11 @@ def run(self, *args, **kwargs):
             for name in self.arg_names:
                 if name in all_args:
                     _args.append(all_args[name])
-            key = tuple(_args[i] for i in self.key_idx)
+            key = [_args[i] for i in self.key_idx]
+            for arg in _args:
```

Comment:
Can we just use the type of those key argument and packet as
[(arg, arg_type(has a type)), (arg, (arg does not have a type)), ... ]

Reply:
I'm not following.
Keys of the autotuner do not have types. The data types added here are from the pointers passed to the kernel.

Reply:
So that means pointer types are always used a part of the key. OK, I see the point here.

Reply:
>> OK, I see the **point** here.
I see what you did there.. :)